include/gridtools/common/demangle.hpp

```diff
@@ -0,0 +1,118 @@
+// core::demangle
+//
+// Copyright 2014 Peter Dimov
+// Copyright 2014 Andrey Semashev
+//
+// Distributed under the Boost Software License, Version 1.0.
+// See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt
+
+# pragma once
+
+#include <string>
+
+// __has_include is currently supported by GCC and Clang. However GCC 4.9 may have issues and
+// returns 1 for 'defined( __has_include )', while '__has_include' is actually not supported:
+// https://gcc.gnu.org/bugzilla/show_bug.cgi?id=63662
+#if defined( __has_include ) && (!defined(__GNUC__) || (__GNUC__ + 0) >= 5)
+# if __has_include(<cxxabi.h>)
+#  define GT_HAS_CXXABI_H
+# endif
+#elif defined( __GLIBCXX__ ) || defined( __GLIBCPP__ )
+# define GT_HAS_CXXABI_H
+#endif
+
+#if defined( GT_HAS_CXXABI_H )
+# include <cxxabi.h>
+// For some archtectures (mips, mips64, x86, x86_64) cxxabi.h in Android NDK is implemented by gabi++ library
+// (https://android.googlesource.com/platform/ndk/+/master/sources/cxx-stl/gabi++/), which does not implement
+// abi::__cxa_demangle(). We detect this implementation by checking the include guard here.
+# if defined( __GABIXX_CXXABI_H__ )
+#  undef GT_HAS_CXXABI_H
+# else
+#  include <cstdlib>
+#  include <cstddef>
+# endif
+#endif
+
+namespace gridtools
+{
+
+namespace core
```

Comment:
This is currently `core` only because that's where boost kept it. Should this go in `common` to match the directory? Or perhaps `impl`/`detail` if it's not meant for public use?

Reply:
I would move to `common`. I think we mention somewhere in the docs that `common` is considered internal, so no need to add impl.